tests/run-make/comment-section/rmake.rs

```diff
@@ -0,0 +1,41 @@
+// Both GCC and Clang write by default a `.comment` section with compiler information.
+// Rustc received a similar .comment section, so this tests checks that this section
+// properly appears.
+// See https://github.com/rust-lang/rust/commit/74b8d324eb77a8f337b35dc68ac91b0c2c06debc
+
+//@ only-linux
+
+use std::path::PathBuf;
+
+use run_make_support::llvm_readobj;
+use run_make_support::rustc;
+use run_make_support::{cwd, env_var, read_dir, run_in_tmpdir};
+
+fn main() {
+    let target = env_var("TARGET");
+    let p = cwd();
+
+    rustc().input("main.rs").emit("link,obj").arg("-Csave-temps").target(&target).run();
+
+    // Check linked output has a `.comment` section with the expected content.
+    llvm_readobj()
+        .gnu_elf_style()
+        .section(".comment")
+        .arg(PathBuf::from(&p).join("main"))
+        .run()
+        .assert_stdout_contains("rustc version 1.");
+
+    // Check all object files (including temporary outputs) have a `.comment`
+    // section with the expected content.
+    read_dir(p, |f| {
+        if f.extension() != Some(".o".as_ref()) {
+            return;
+        }
```

Comment:
Problem: I believe this check is not correct, if an object artifact is called `foo.o`, its `extension()` is `o`, the period is not included.

```rs
fn main() {
    let path = std::path::PathBuf::from("foo.o");
    println!("{:?}", path.extension())
}

// outputs: `Some("o")`
```

Suggestion: since there is an PartialEq impl between `OsStr` and `str`, you can also just write

```rs
if f.extension().is_some_and(|ext| ext == "o") {
    return;
}
```